# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  a95d27030e93ca65276bf9b1769f0ba6053c41e6
Bug 2073417 - Fix broken offline for POP AGAIN. Follow-up to unfortunate TB work in bug 2071585.

The suggested hunk was already in our earlier patch now at
https://github.com/Betterbird/thunderbird-patches/blob/199f2eb60205bf6cad7c1d578d91739c31c3ddc1/153/unused/2071585-fix-offline-for-pop.patch#L54
which the TB folks chose to ignore.

diff --git a/mailnews/local/src/Pop3IncomingServer.sys.mjs b/mailnews/local/src/Pop3IncomingServer.sys.mjs
--- a/mailnews/local/src/Pop3IncomingServer.sys.mjs
+++ b/mailnews/local/src/Pop3IncomingServer.sys.mjs
@@ -114,8 +114,11 @@ export class Pop3IncomingServer extends 
     return MailServices.pop3.verifyLogon(this, urlListener, msgWindow);
   }
 
   performBiff(msgWindow) {
+    if (Services.io.offline) {
+      return;
+    }
     this.performingBiff = true;
     const inbox = this.rootMsgFolder.getFolderWithFlags(
       Ci.nsMsgFolderFlags.Inbox
     );
